File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        obj=defaultdict(bool)
        for i in range(len(nums)):
            if not obj[nums[i]]:
                obj[nums[i]]=[i]
            else:
                obj[nums[i]].append(i)
        nums.sort()
        
        for i in range(len(nums)):
            if i+1<len(nums):
                if nums[i]==nums[i+1]:
                    for j in range(len(obj[nums[i]])):
                        if j+1<len(obj[nums[i]]):
                            if abs(obj[nums[i]][j+1]-obj[nums[i]][j])<=k:
                                return True
        return False
        

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
    let obj={}
    for(let i=0;i<nums.length;i++){
        if(!obj[nums[i]]){
            obj[nums[i]]=[i]
        }else{
            obj[nums[i]].push(i)
        }    
    }
     // console.log(obj)
    for(let i in obj){
        // console.log(obj[i])
        if(obj[i].length>1){
            for(let j=0;j<obj[i].length;j++){
                if(obj[i][j+1]-obj[i][j]<=k){
                    return true
                }
            }
        } 
    }
    return false
   
};